packages/core/docs/integrate/integration-guide.md

```diff
@@ -0,0 +1,369 @@
+
+
+## 1. Creating an api-client
+
+Each integration starts with `api-client`. This is one of the packages which is responsible for communication between the Vue Storefront and external API. That's exactly the place where you have to configure your api connection, write your api functions, and expose generated api client to the users.
+
+### 1.1 Configuration
+The creation of api client starts with the configuration:
+
+```ts
+// api-client/src/configuration.js
+import getProduct from './api/getProduct';
+import getCategory from './api/getCategory';
+import { apiClientFactory } from '@vue-storefront/core';
+
+interface Config {
+  // ...
+}
+
+interface API {
+  // ..
+}
+
+interface OnSetup {
+  config: Config;
+  client: EcommerceAPI;
+}
+
+const onSetup = (config: Config): OnSetup => {
+  const client = new EcommerceAPI(config)
+
+  return {
+    config,
+    client
+  };
+};
+
+const { createApiClient } = apiClientFactory<Config, API>({
+  tag: 'ct',
+  onSetup,
+  api: {
+    getProduct,
+    getCategory,
+  }
+});
+```
+
+```ts
+// api-client/src/index.js
+import { createApiClient } from './configuration';
+
+export { createApiClient }
+```
+
+To create api-client you need to call `apiClientFactory`. As a parameters to this deliver a few things:
+- `tag` - that's the short name of your integration which will be used to distinguish it among of others
+- `onSetup` - optional function that will be called during the creating your api. In this place you can call everything you need to create a connection to the api, such as creating sdk (eg. axios creation), merge given config with the defaults etc.
+- `api` - this is the section where you need to pass all of the api function you have created
+
+### 1.2 Api functions
+Once you have your configuration created, you can proceed with api functions:
+
+```ts
+// api-client/src/api/getProduct.js
+const getProduct = (context, params) => {
+  const productResponse = await context.client.get(`/product/${params.id}`);
+
+  return productResponse.data
+}
+```
+
+Each api function always contains `context` as a first parameter. This is the place where you always have access to the client and config of your api connection. Obviously, during the using api client, you will be using these functions without thinking about the context - the vsf core handles this. At the end, you need to provide that function to the api creation logic (section above)
+
+## 2. Creating composables
+
+Composables are the major part of integration. That exactly the place where the business logic comes in. We always serving this package as a integration along with corresponding nuxt module.
+
+### 2.1 Exposing integration plugin wrapper
+
+The first thing that you do, is exposing a nuxt plugin wrapper for creation and configuring your integration by nuxt environment. Within your composables package you have to add these few lines:
```

Comment:
Let's make a convention about naming libraries with capitalized first letter (except webpack and npm 😅 )
```suggestion
The first thing that you do, is exposing a Nuxt plugin wrapper for the creation and configuring your integration by Nuxt environment. Within your composables package you have to add these few lines:
```